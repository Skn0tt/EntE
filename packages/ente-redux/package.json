{
  "name": "ente-redux",
  "version": "0.2.3",
  "description": "State Management for EntE",
  "main": "src/index.ts",
  "repository": "gitlab.com/skn0tt/EntE/master/packages/ente-redux",
  "license": "AGPL-3.0-or-later",
  "author": {
    "email": "info@simonknott.de",
    "name": "Simon Knott",
    "url": "https://simonknott.de"
  },
  "private": true,
  "dependencies": {
    "axios": "^0.18.0",
    "ente-lang": "^0.2.3",
    "ente-types": "^0.2.3",
    "immutable": "^3.8.2",
    "redux": "^3.7.2",
    "redux-actions": "^2.3.0",
    "redux-saga": "^0.16.0",
    "reselect": "^3.0.1"
  },
  "devDependencies": {
    "@types/redux-actions": "^2.2.4"
  }
}
