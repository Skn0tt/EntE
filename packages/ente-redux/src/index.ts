/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import { createStore } from "./store";
import { Middleware } from "redux";

export * from "./selectors";
export * from "./actions";
export * from "./types";
export * from "./constants";

export type ReduxConfig = {
  baseUrl: string;
  middlewares?: Middleware[];
  onSagaError?: (err: Error) => void;
  onFileDownload: (file: Blob, filename: string) => void;
};

export let config: ReduxConfig = {
  baseUrl: "",
  middlewares: [],
  onFileDownload: () => {}
};

const setup = (conf: ReduxConfig) => {
  config = conf;

  return createStore();
};

export default setup;
