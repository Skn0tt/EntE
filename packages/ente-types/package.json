{
  "name": "ente-types",
  "version": "0.15.0",
  "description": "Typings for EntE",
  "main": "src/index.ts",
  "repository": "gitlab.com/skn0tt/EntE/master/ente-types",
  "license": "AGPL-3.0-or-later",
  "author": {
    "email": "info@simonknott.de",
    "name": "Simon Knott",
    "url": "https://simonknott.de"
  },
  "private": true,
  "scripts": {
    "build": "yarn tsc"
  },
  "devDependencies": {
    "typescript": "^2.8.1"
  },
  "dependencies": {
    "class-transformer": "0.1.10",
    "class-validator": "^0.9.1",
    "date-fns": "^2.0.0-alpha.26"
  }
}
