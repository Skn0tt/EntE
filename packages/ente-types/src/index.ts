/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

export * from "./roles";
export * from "./dtos";
export * from "./validators";
export * from "./helpers";
