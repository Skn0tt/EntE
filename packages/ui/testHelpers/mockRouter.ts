import { RouteComponentProps } from "react-router";
import { Location, UnregisterCallback } from "history";

export function getMockRouterProps<P>(data: P) {
  const location: Location = {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  };

  const props: RouteComponentProps<P> = {
    location,
    match: {
      isExact: true,
      params: data,
      path: "",
      url: ""
    },
    history: {
      location,
      length: 2,
      action: "POP",
      push: () => {},
      replace: () => {},
      go: num => {},
      goBack: () => {},
      goForward: () => {},
      block: t => null as any,
      createHref: t => "",
      listen: t => null as any
    },
    staticContext: {}
  };

  return props;
}
