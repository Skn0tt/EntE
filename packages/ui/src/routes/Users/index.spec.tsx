import * as React from "react";
import { Users } from "./index";
import { shallow } from "enzyme";
import { User } from "ente-redux";
import { getMockRouterProps } from "../../../testHelpers/mockRouter";
import { Roles } from "ente-types";

describe("Users", () => {
  const users: User[] = [
    new User({
      username: "simon",
      displayname: "Simon",
      email: "email@emai.com",
      isAdult: false,
      role: Roles.STUDENT,
      _id: "mystupidid"
    })
  ];
  const getUsers = jest.fn();
  const comp = shallow(
    <Users
      {...getMockRouterProps()}
      classes={{}}
      users={users}
      getUsers={getUsers}
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  it("requests all users", () => {
    expect(getUsers).toBeCalled();
  });
});
