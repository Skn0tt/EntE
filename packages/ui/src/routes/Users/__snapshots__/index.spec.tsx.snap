// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Users
    classes={Object {}}
    getUsers={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 2,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "",
          "pathname": "",
          "search": "",
          "state": Object {},
        },
        "push": [Function],
        "replace": [Function],
      }
    }
    location={
      Object {
        "hash": "",
        "key": "",
        "pathname": "",
        "search": "",
        "state": Object {},
      }
    }
    match={
      Object {
        "isExact": true,
        "params": undefined,
        "path": "",
        "url": "",
      }
    }
    staticContext={Object {}}
    users={
      Array [
        Immutable.User {
          "_id": "mystupidid",
          "username": "simon",
          "displayname": "Simon",
          "email": "email@emai.com",
          "role": "student",
          "isAdult": false,
          "children": Array [],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(WithTheme(WithWidth(WithMobileDialog)))
          onClose={[Function]}
          show={false}
        />,
        <WithStyles(Table)
          cellExtractor={[Function]}
          headers={
            Array [
              "Username",
              "Displayname",
              "Email",
              "Role",
            ]
          }
          items={
            Array [
              Immutable.User {
                "_id": "mystupidid",
                "username": "simon",
                "displayname": "Simon",
                "email": "email@emai.com",
                "role": "student",
                "isAdult": false,
                "children": Array [],
              },
            ]
          }
          keyExtractor={[Function]}
          onClick={[Function]}
        />,
        <WithStyles(Button)
          className={undefined}
          color="primary"
          onClick={[Function]}
          variant="fab"
        >
          <pure(Add) />
        </WithStyles(Button)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "onClose": [Function],
          "show": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cellExtractor": [Function],
          "headers": Array [
            "Username",
            "Displayname",
            "Email",
            "Role",
          ],
          "items": Array [
            Immutable.User {
              "_id": "mystupidid",
              "username": "simon",
              "displayname": "Simon",
              "email": "email@emai.com",
              "role": "student",
              "isAdult": false,
              "children": Array [],
            },
          ],
          "keyExtractor": [Function],
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <pure(Add) />,
          "className": undefined,
          "color": "primary",
          "onClick": [Function],
          "variant": "fab",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(WithTheme(WithWidth(WithMobileDialog)))
            onClose={[Function]}
            show={false}
          />,
          <WithStyles(Table)
            cellExtractor={[Function]}
            headers={
              Array [
                "Username",
                "Displayname",
                "Email",
                "Role",
              ]
            }
            items={
              Array [
                Immutable.User {
                  "_id": "mystupidid",
                  "username": "simon",
                  "displayname": "Simon",
                  "email": "email@emai.com",
                  "role": "student",
                  "isAdult": false,
                  "children": Array [],
                },
              ]
            }
            keyExtractor={[Function]}
            onClick={[Function]}
          />,
          <WithStyles(Button)
            className={undefined}
            color="primary"
            onClick={[Function]}
            variant="fab"
          >
            <pure(Add) />
          </WithStyles(Button)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onClose": [Function],
            "show": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cellExtractor": [Function],
            "headers": Array [
              "Username",
              "Displayname",
              "Email",
              "Role",
            ],
            "items": Array [
              Immutable.User {
                "_id": "mystupidid",
                "username": "simon",
                "displayname": "Simon",
                "email": "email@emai.com",
                "role": "student",
                "isAdult": false,
                "children": Array [],
              },
            ],
            "keyExtractor": [Function],
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <pure(Add) />,
            "className": undefined,
            "color": "primary",
            "onClick": [Function],
            "variant": "fab",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
